solutions/usaco-1017.mdx

```diff
@@ -15,6 +15,76 @@ author: Sofia Yang
 
 <LanguageSection>
 
+<CPPSection>
+
+```cpp
+#include <bits/stdc++.h>
+typedef long long ll;
+using namespace std;
+
+//Construct a topologically sorted array from this function
+void dfs(vector<vector<pair<int,int>>>& graph, vector<bool>& visited, vector<int>& toposort, int node){
+	visited[node] = true;
+	for (auto i: graph[node]){
+		int a,b;
+		tie(a,b) = i;
+		if (visited[a]) continue;
+		dfs(graph, visited, toposort, a);
+	}
+	toposort.push_back(node);
+}
+
+
+int main() {
+	ofstream fout("timeline.out");
+	ifstream fin("timeline.in");
+
+	//Take inputs
+	//Construct an array that track all the start time
+	//Construct an adjacency list graph
+	int n, m, c;
+	fin >> n >> m >> c;
+	vector<int> start(n+1);
+	for (int i = 1; i <= n; i++){
+		fin >> start[i];
+	}
+	vector<vector<pair<int, int>>> graph(n + 1);
+	for (int i = 0; i < c; i++){
+		int a, b, c;
+		fin >> a >> b >> c;
+		graph[a].push_back(make_pair(b,c));
```

Comment:
(optional) make nodes 0-indexed

Reply:
ok, 1-indexing makes the code looks bit nicer, no need to put `-1` everywhere

Reply:
really? it's just a `--` to all the indices?

Reply:
I mean, is it really necessary to write `--` everywhere when you don't have to

Reply:
everywhere?

Reply:
bump